486 grant permissions registration app

Sepehr-Sobhani opened the pull request:
[ISSUE 486](https://github.com/bcgov/cas-reporting/issues/486)

**NOTES:**
- Added grants for the registration app.
- Grants access to the public schema when developing locally to enable record creation in the Silk Profiler database.
- Some tests from Reg1 are skipped or deleted because the logic from Reg1 does not align with the grants, so I didn't want to spend time fixing them.
- Applying RLS whenever we run the migrate command (post-migrate signal).
- Added grants to historical m2m fields manually (the Django Simple History package autogenerates these tables, and we lack control over the schema they are generated in).
- Passed `user_guid` directly to some functions to enable allowed internal users to create contacts (internal users should be able to create a contact but not update one).
- Passed `user_guid` directly to some functions to enable allowed internal users to create `bcghg_id` and `boro_id` (internal users should be able to create them but not update them. We need to pass `user_guid` to populate `issued_by_id` during creation).


**TEST:(Developer only)**
- Set `RLS_FLAG = True` in `settings.py` file
- Change `if settings.RLS_FLAG is False:` to `if settings.RLS_FLAG is True:` in `bc_obps/rls/models.py`

BCerki reviewed (commented):
No questions about the code since the walkthrough was thorough. The following came up in manual testing:
- I'm no longer able to approve/deny access requests as cas_director: 
![image](https://github.com/user-attachments/assets/d1c65fc1-096b-4145-87f4-1219047ce6ae)
- I had trouble doing this transfer as a cas_analyst: 
![image](https://github.com/user-attachments/assets/7baec89d-856a-4131-8e99-7cba414015a2)
Then when I changed the effective transfer date and tried to submit, I got this: 
![image](https://github.com/user-attachments/assets/3d00d80c-26ec-4331-8aa8-c2c631fb78de), so I think the error is beyond create_transfer. If I try to edit a transfer to put the effective date in the future, that also fails with the same error.

-This is unrelated to your PR I think, but if I edit this transfer: 
![image](https://github.com/user-attachments/assets/052f6676-80ca-4b9f-9140-6432a6d90fc7)
And add Facility 7, I get an error, but the form goes read-only and it looks like it worked: 
![image](https://github.com/user-attachments/assets/0ea2ef08-22e3-46e2-b4f5-776998129ae1)
. On refresh Facility 7 is gone

Sepehr-Sobhani commented:
> No questions about the code since the walkthrough was thorough. The following came up in manual testing:
> 
> * I'm no longer able to approve/deny access requests as cas_director:
>   ![image](https://private-user-images.githubusercontent.com/77306764/413442118-d1c65fc1-096b-4145-87f4-1219047ce6ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk1ODE1MTIsIm5iZiI6MTczOTU4MTIxMiwicGF0aCI6Ii83NzMwNjc2NC80MTM0NDIxMTgtZDFjNjVmYzEtMDk2Yi00MTQ1LTg3ZjQtMTIxOTA0N2NlNmFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDAxMDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0ZDI5MmJiNjNmYjkwMDAzZjkzODJjZWJlN2M0ZTJlNGU0ZmJmZTU1YTc3ZTY3NDI2ZGQxYjdlODQzMjI2MDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xXR2m8LufELVN95LTR_B1hmLSRB07nHCu2gn1fLusJA)
> * I had trouble doing this transfer as a cas_analyst:
>   ![image](https://private-user-images.githubusercontent.com/77306764/413446347-7baec89d-856a-4131-8e99-7cba414015a2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk1ODE1MTIsIm5iZiI6MTczOTU4MTIxMiwicGF0aCI6Ii83NzMwNjc2NC80MTM0NDYzNDctN2JhZWM4OWQtODU2YS00MTMxLThlOTktN2NiYTQxNDAxNWEyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDAxMDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczODU0YWNhYTVjZGEzMTU3YjQ4OWRmMDdkZTlkZjdlYjllMGE2ZWE2ZjgxNDliMDhjMTJmNWM4ZDFlNTI2NmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-wycFkJFmVY3UJ3KSPDhKKBKlYsEDGzqTYaC-Uowx2s)
>   Then when I changed the effective transfer date and tried to submit, I got this:
>   ![image](https://private-user-images.githubusercontent.com/77306764/413446586-3d00d80c-26ec-4331-8aa8-c2c631fb78de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk1ODE1MTIsIm5iZiI6MTczOTU4MTIxMiwicGF0aCI6Ii83NzMwNjc2NC80MTM0NDY1ODYtM2QwMGQ4MGMtMjZlYy00MzMxLThhYTgtYzJjNjMxZmI3OGRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDAxMDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzMGJkZTczZWRkMTZhYjg3NDg3MWMzYjQwYTMzYjMzNmFiMDUxMTE5MDU2YmNiODBjYTc0OGU3ZDhjYTk3YzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RhCFqfzAg1RqqFZ1DVgqjObDy6iLJJn_EZZ4qDWEJXc), so I think the error is beyond create_transfer. If I try to edit a transfer to put the effective date in the future, that also fails with the same error.
> 
> -This is unrelated to your PR I think, but if I edit this transfer: ![image](https://private-user-images.githubusercontent.com/77306764/413447285-052f6676-80ca-4b9f-9140-6432a6d90fc7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk1ODE1MTIsIm5iZiI6MTczOTU4MTIxMiwicGF0aCI6Ii83NzMwNjc2NC80MTM0NDcyODUtMDUyZjY2NzYtODBjYS00YjlmLTkxNDAtNjQzMmE2ZDkwZmM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDAxMDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2ZWExNTU1MGIwNDM4YjIxY2ZhNTdhZDRhNGYyYzQ3YmUxYTk0YWJmZTE5MWU5YTEzNWYzMzZlMjQ3NDUyOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sE2QbkoNbFH1bGgAa-s_4B_vMOu-65pQMBVw8odo_lc) And add Facility 7, I get an error, but the form goes read-only and it looks like it worked: ![image](https://private-user-images.githubusercontent.com/77306764/413447453-0ea2ef08-22e3-46e2-b4f5-776998129ae1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk1ODE1MTIsIm5iZiI6MTczOTU4MTIxMiwicGF0aCI6Ii83NzMwNjc2NC80MTM0NDc0NTMtMGVhMmVmMDgtMjJlMy00NmUyLWI0ZjUtNzc2OTk4MTI5YWUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE1VDAxMDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwM2I0OGEwOWM0ZjljM2RiYzg3MWIyNjEzY2NlNTE0NWJlNDZiYTViZGFkNGRlNTlmYTlkNTBkMDZhYzk4YTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.K16AdnlQ8lhYLzazgq0l5Jw96GAEtB6J4X9Bn2o_wCM) . On refresh Facility 7 is gone


Thank you for reviewing this PR!

I've added two missing grants for the Facility and Operation data models, which I believe were causing related transaction errors. Since a transaction error can block others, this issue may have led to errors in other forms as well. My changes should resolve these issues, but please let me know if anything still needs fixing.

I also updated the endpoint exception handler to return Internal Server Error instead of the transaction error. Additionally, I wrapped the event creation process in a transaction atomic block to ensure an event isn't created if it can't be processed.

dleard reviewed (approved):
Great work! The grants & supporting arcthitecture look good. Approved on my end, but let's wait to merge until Brianna confirms the issues she found have been addressed

BCerki commented on bc_obps/service/tests/test_operation_service_v2.py:
```diff
@@ -1260,9 +1260,10 @@ def test_raises_exception_if_operation_is_non_regulated():
 
         with pytest.raises(Exception, match="Non-regulated operations cannot be issued BORO ID."):
-            OperationServiceV2.generate_boro_id(approved_user_operator.user.user_guid, operation.id)
+            OperationServiceV2.generate_boro_id(cas_director.user_guid, operation.id)
```
good call

BCerki reviewed (approved):
Nice work! :cartwheeling:
